### PR TITLE
Clean up dynamic extension CLI

### PR DESCRIPTION
**Pull request by blink1073**
## References
Fixes https://github.com/jupyterlab/jupyterlab-module-federation/issues/46

## Code changes
- [x] `labextension develop`:
  -  Add to python path if it doesn't exist
  - Run build if necessary - if there are no static assets
- [x] `labextension build/watch` -  clear the contents of the folder, do not remove the folder
- [x] Add dependency on `webpack-cli` and loaders in `buildutils`, since they are needed for extension authors
   - Verified against https://github.com/jupyterlab/extension-cookiecutter-ts/pull/73 using `yarn link`

## User-facing changes
Ease of use for extension authors

## Backwards-incompatible changes
None

### CODE REVIEW

**Comment by blink1073** [comment]
We will need another alpha release before I can finish https://github.com/jupyterlab/extension-cookiecutter-ts/pull/73

File: buildutils/src/ensure-repo.ts [hunk 1]
```diff
@@ -37,7 +37,19 @@ const UNUSED: Dict<string[]> = {
...
+    'html-loader',
+    'raw-loader',
+    'source-map-loader',
+    'style-loader',
+    'svg-url-loader',
+    'terser-webpack-plugin',
+    'to-string-loader',
+    'url-loader',
+    'webpack-cli',
+    'worker-loader'
```
Comment (jasongrout) [inline buildutils/src/ensure-repo.ts:52]: You added a lot more packages than I did: https://github.com/jupyterlab/jupyterlab/pull/8850/commits/c4effe0900876a5afbc8340964f2ef0b29a7fc55

I just added exactly the packages I needed to get the example to compile. Are we sure that we need each one of these loaders, and terser?

Comment (blink1073) [reply buildutils/src/ensure-repo.ts:52]: We need all the ones in the [base config](https://github.com/jupyterlab/jupyterlab/blob/master/buildutils/src/webpack.config.base.ts), and we'll need terser once we support production builds for extensions.

Comment (blink1073) [reply buildutils/src/ensure-repo.ts:52]: We might not need `worker-loader`, but we added that due to build errors folks were seeing.

Comment (blink1073) [reply buildutils/src/ensure-repo.ts:52]: We also might not explicitly need `to-string-loader`, since right now that is only used for codemirror.

Comment (jasongrout) [reply buildutils/src/ensure-repo.ts:52]: I needed to-string-loader when one of the example extensions imported codemirror, even if it didn't actually bundle codemirror.

**Comment by jasongrout** [comment]
I'll fix the conflicts with my PR over on my PR. Thanks.

**Pull request merged**